doc/source/rllib/package_ref/replay_buffers.rst

```diff
@@ -0,0 +1,132 @@
+.. _replay-buffer-api-reference-docs:
+
+Replay Buffer API
+==================
+
+RLlib comes with a set of extendable replay buffers which are used mostly in Q-Learning algorithms.
+We provide a base :py:class:`~ray.rllib.utils.replay_buffers.replay_buffer.ReplayBuffer` class from which you can build your own buffer.
+Some use cases may require :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_replay_buffer.MultiAgentReplayBuffer`\s.
```

Comment:
Can we specify "some"? That should mean in "multi-agent scenarios", right?

Reply:
We use the MultiAgentReplayBuffers for almost everything. Only SimpleQ uses an ordinary replay buffer.
For DQN, R2D2 etc. the multi agent buffers are not really needed, but have been used historically. Do you think we should think about undoing this? It's not something I came up with, I just view them as a generalization from the "single-agent" buffers.

Reply:
Yeah, maybe we should explain this here. We chose `MultiAgentBuffer` as default b/c we don't want the user to have to switch to it only b/c they are configuring their problem as a multi-agent one.
I do think, we should also use MultiAgentBuffer for DQN and R2D2 (and even SimpleQ). There is no good reason not to. Otherwise, setting up these algos for multi-agent requires an additional step, which is hard to explain to the users.

Reply:
Just realized that this has been changed already!